segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -173,6 +211,20 @@ public void initChannel(SocketChannel ch) throws Exception {
              }
          });
 
+        if (tlsEnabled && tlsReloadEnabled) {
+            log.debug("TLS reload is enabled, so setting up a FileChangeWatcher object to watch changes in file: {}",
+                    this.pathToTlsCertFile);
+
+            FileModificationWatcher tlsCertFileChangeWatcherTask = new FileModificationWatcher(
+                    this.pathToTlsCertFile,
+                    new TLSConfigChangeEventConsumer(channels));
+            tlsCertFileChangeWatcherTask.setDaemon(true);
+
+            this.executor.submit(tlsCertFileChangeWatcherTask);
```

Comment:
So you have a `FileModificationWatcher` which extends `Thread`, which you then submit to an `ExecutorService` which has its own threads. That's one thread too many. 

I suggest you remove the executor from this class and simply execute your one-off thread. But if you do that, you need to keep a reference to it so that when you do close this class, you interrupt that thread.

Reply:
Yes, I agree. I've changed it in the latest commit.